File: EnterValues.js
// configEditor.js
import fs from 'fs';
import readline from 'readline';
import chalk from 'chalk';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const CONFIG_PATH = './src/config.json';

function prompt(question) {
  return new Promise(resolve => rl.question(chalk.cyan(question), answer => resolve(answer)));
}

async function editConfig() {
  console.clear();
  console.log(chalk.yellowBright("=== GhostUsernameHunterPY - Config Editor ===\n"));

  let config = {};
  if (fs.existsSync(CONFIG_PATH)) {
    config = JSON.parse(fs.readFileSync(CONFIG_PATH, 'utf-8'));
  } else {
    console.log(chalk.red("No config.json file found. Creating a new one.\n"));
  }

  config.url = await prompt(`Server URL [${config.url || ''}]: `) || config.url;
  config.logout_url = await prompt(`Logout URL [${config.logout_url || ''}]: `) || config.logout_url;
  config.method = (await prompt(`Method (GET or POST) [${config.method || 'POST'}]: `)).toUpperCase() || config.method || 'POST';
  config.length = parseInt(await prompt(`Total Username Length [${config.length || 8}]: `)) || config.length || 8;
  config.digits = await prompt(`Digits for Random Part [${config.digits || '0123456789'}]: `) || config.digits || '0123456789';
  config.prefix = await prompt(`Username Prefix [${config.prefix || ''}]: `) || config.prefix || '';
  config.suffix = await prompt(`Username Suffix [${config.suffix || ''}]: `) || config.suffix || '';
  config.count = parseInt(await prompt(`Usernames to Generate [${config.count || 50}]: `)) || config.count || 50;

  fs.writeFileSync(CONFIG_PATH, JSON.stringify(config, null, 2), 'utf-8');

  console.log(chalk.greenBright("\nConfiguration updated successfully!\n"));
  rl.close();
}

editConfig();

File: enter_values.py
#!/usr/bin/env python3
"""Python port of EnterValues.js - interactive config.json editor."""

import json
import os

from colorama import Fore, init as colorama_init

colorama_init(autoreset=True)

CONFIG_PATH = os.path.join(os.path.dirname(__file__), "src", "config.json")


def prompt(question):
    return input(Fore.CYAN + question)


def edit_config():
    os.system("cls" if os.name == "nt" else "clear")
    print(Fore.YELLOW + "=== GhostUsernameHunterPY - Config Editor ===\n")

    config = {}
    if os.path.exists(CONFIG_PATH):
        with open(CONFIG_PATH, "r", encoding="utf-8") as f:
            config = json.load(f)
    else:
        print(Fore.RED + "No config.json file found. Creating a new one.\n")

    config["url"] = prompt(f"Server URL [{config.get('url', '')}]: ") or config.get("url", "")
    config["logout_url"] = prompt(f"Logout URL [{config.get('logout_url', '')}]: ") or config.get("logout_url", "")
    config["method"] = (prompt(f"Method (GET or POST) [{config.get('method', 'POST')}]: ").upper()
                         or config.get("method", "POST"))

    length_input = prompt(f"Total Username Length [{config.get('length', 8)}]: ")
    try:
        config["length"] = int(length_input)
    except ValueError:
        config["length"] = config.get("length", 8)

    config["digits"] = prompt(f"Digits for Random Part [{config.get('digits', '0123456789')}]: ") or config.get("digits", "0123456789")
    config["prefix"] = prompt(f"Username Prefix [{config.get('prefix', '')}]: ") or ""
    config["suffix"] = prompt(f"Username Suffix [{config.get('suffix', '')}]: ") or ""

    count_input = prompt(f"Usernames to Generate [{config.get('count', 50)}]: ")
    try:
        config["count"] = int(count_input)
    except ValueError:
        config["count"] = config.get("count", 50)

    with open(CONFIG_PATH, "w", encoding="utf-8") as f:
        json.dump(config, f, indent=2, ensure_ascii=False)

    print(Fore.GREEN + "\nConfiguration updated successfully!\n")


if __name__ == "__main__":
    edit_config()
